js/controller.js

```diff
@@ -275,6 +275,7 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
 
     onEmbedCodeChanged: function(event, embedCode, options) {
       this.state.videoQualityOptions.availableBitrates = null;
+      this.state.closedCaptionOptions.availableLanguages = null;
```

Comment:
From what I see, every time we switch to a new video, we are getting OO.EVENTS.DISCOVERY_API.RELATED_VIDEOS_FETCHED which replaces discoveryData. I was thinking in case no related videos available for some reason, we shouldn't be reusing the discovery data from the previous video.